src/IJulia.jl

```diff
@@ -211,6 +211,36 @@ a subset of cell inputs/outputs to clear.
 clear_history
 
 #######################################################################
+# methods to print history or any subset thereof
+function history(io::IO, indices)
+	# since a range could be huge, intersect it with 1:n first
+    for n in intersect(indices, 1:IJulia.n)
+      if haskey(In, n)
+        print(In[n])
+      end
+    end
+end
+history(x) = history(Base.STDOUT, x)
+history() = history(1:n)
+
+# print the last x history entries when argument x is integer
+history(io::IO, x::Integer) =
+    invoke(history, Tuple{IO, Any}, io, n-x:n)
```

Comment:
(We aren't bound to emulate IPython's behavior, of course, but it's a hint about what people have found useful.)